server/controllers/users.js

```
@@ -0,0 +1,324 @@
+      false;
+    const password = Validation.checkPasswordValidityOf(req.body.password) ?
+      Validation.checkPasswordValidityOf(req.body.password) :
+      false;
+    if(!email || !password) {
+      return res.status(400).json({ message: 'Email and password input cannnot be empty' });
+    }
+    return User.findOne({
+      where: {
+        email
+      }
+    }).then((user) => {
+      const result = bcrypt.compareSync(password, user.password);
+      if(result) {
+        const token = jwt.sign({
+          userId: user.id,
+          userEmail: user.email,
+          userRole: user.roleId
+        }, 'zabuzatovadase', {
+          expiresIn: '24h'
+        });
+        localStorage.set('token', token);
+        res.status(200).json({ user, token });
+      } else {
+        res.status(400).json({ message: 'Invalid Password' });
+      }
+    }).catch((error) => {
+      res.status(500).json({ message: 'Problems with either the email or password, Check and try again' });
+    })
+  }
+  /**
+   * Shows a detail of all the users successfully signed up on the 
+   * database
+   * @param {any} req request made from the client
+   * @param {any} res response from the server
+   * @returns response object
+   */
+  static getUsers(req, res) {
+    const userList = [];
+    const userDetails = req.decoded;
+    if (userDetails.userRole !== Constants.ADMIN && userDetails.userRole !== Constants.SUPERADMIN) {
+      return res.status(403).json({ message: 'You do not have the permission to perform this action' });
+    }
+    if(req.query) {
+      const max = Constants.MAXIMUM;
+      const selectedUsersList = [];
+      const offset = req.query.offset || 0,
+            limit = req.query.limit || max;
+      return User.findAndCountAll({ offset, limit })
+      .then((users) => {
+        if(users.rows.length === 0) {
+          res.status(500).json({ message: 'No users available for the page selected' });
```

What do you think of returning an empty array with a `200` status code if there is no user at the selected page? Or if you really need to return an error, how about a `404` status code?